doc/introduction/data.rst

```diff
@@ -0,0 +1,172 @@
+.. role:: html(raw)
+   :format: html
+
+.. _intro_ref_data:
+
+Quantum Datasets
+================
+
+PennyLane provides the :mod:`~.data` subpackage to download, create, store and manipulate quantum datasets.
+
+Loading Datasets in PennyLane
+-----------------------------
+
+We can access data of a desired type with the :func:`~pennylane.data.load` or :func:`~.data.load_interactive` functions.
+These download the desired datasets or load them from local storage if previously downloaded. 
+Here we use the :func:`~pennylane.data.load` function, for the
+:func:`~.data.load_interactive` function, please see the relevant documentation.
+
+To specify the dataset to be loaded, the data category (``data_name``) must be
+specified, alongside category-specific keyword arguments. For the full list
+of available datasets, please see the `datasets website <https://pennylane.ai/qml/datasets.html>`_.
+The :func:`~pennylane.data.load` function returns a list with the desired data.
+
+.. code-block:: python
+
+    >>> H2_dataset = qml.data.load(data_name="qchem", molname="H2", basis="STO-3G", bondlength="1.0")
+    >>> print(H2_dataset)
+    [<pennylane.data.dataset.Dataset object at 0x7f14e4369640>]
+
+When we only want to download portions of a large dataset, we can specify the desired properties  (referred to as `attributes`).
+For example, we can download or load only the molecule and Hamiltonian of a dataset as follows:
+
+.. code-block:: python
+
+    >>> H2_partial = qml.data.load(data_name='qchem',molname='H2', basis='STO-3G', bondlength=1.0, attributes=['molecule','fci_energy'])[0]
+    >>> H2_partial.molecule
+    <pennylane.qchem.molecule.Molecule at 0x7f56c9d78e50>
+    >>> H2_partial.fci_energy
+    -1.1011498981604342
+
+To determine what attributes are available for a type of dataset, we can use the function :func:`~pennylane.data.list_attributes`:
+
+.. code-block:: python
+
+    >>> qml.data.list_attributes(data_name='qchem')
+    ['molecule',
+    'hamiltonian',
+    'wire_map',
+    ...
+    'vqe_params',
+    'vqe_circuit']
+
+Using Datasets in PennyLane
+---------------------------
+
+Once loaded, one can access properties of the datasets:
+
+.. code-block:: python
+
+    >>> H2_dataset[0].molecule
+    <pennylane.qchem.molecule.Molecule object at 0x7f890b409280>
+    >>> print(H2_dataset[0].hf_state)
+    [1 1 0 0]
```

Comment:
Note: this is technically _not_ `python` syntax, but `pycon` syntax, so you should change this (and other examples) to

```rest
.. code-block:: pycon
```

_However_, the nice thing about ReST is that it recognizes pycon syntax automatically! So you don't even need to use `.. code-block::` and indent.

For example:

```rest
Once loaded, one can access properties of the datasets:

>>> H2_dataset[0].molecule
<pennylane.qchem.molecule.Molecule object at 0x7f890b409280>
>>> print(H2_dataset[0].hf_state)
[1 1 0 0]
```